core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -304,6 +305,17 @@ private SqlLibraryOperators() {
           OperandTypes.CHARACTER,
           SqlFunctionCategory.STRING);
 
+  @LibraryOperator(libraries = {MYSQL})
+  public static final SqlFunction MAP_FILTER =
+      new SqlFunction("MAP_FILTER",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.cascade(ReturnTypes.ARG0, SqlTypeTransforms.TO_NULLABLE),
+          null,
+          OperandTypes.sequence("MAP_FILTER(<ANY>, <LAMBDA(BOOLEAN, ANY, ANY)>)",
+              OperandTypes.family(SqlTypeFamily.MAP),
+              OperandTypes.lambda(SqlTypeFamily.BOOLEAN, SqlTypeFamily.ANY, SqlTypeFamily.ANY)),
+          SqlFunctionCategory.SYSTEM);
```

Comment:
Shouldn't the operand type check be `OperandTypes.family(SqlTypeFamily.MAP, SqlTypeFamily.LAMBDA)` ?

Reply:
The Map Function accepts lambda which returns boolean value and takes two arguments of any type. This seems to be correct.